### ISSUE TITLE
Undeclared identifier errors 

### ISSUE THREAD

**Issue by zetashift** [pragmagic/godot-nim#36]
So this works: 
```nim
import godot, node_2d, rich_text_label, line_edit, strformat

gdobj Loony of Node2D:
  var storyText: RichTextLabel
  var textBox: LineEdit
  var prompts = @["Rishi", "lice", "bad", "great", "cabbages"]
  var story = &"Power"

  method ready*() =
    storyText = getNode("Blackboard/StoryText").as(RichTextLabel)
    storyText.text = story

  method onTextEntered*(newText: string) {.base.} =
    storyText = getNode("Blackboard/StoryText").as(RichTextLabel)
    storyText.text = newText
    textBox = getNode("Blackboard/TextBox").as(LineEdit)
    textBox.clear()

  method onButtonPressed*() {.base.} =
    var newText = getNode("Blackboard/TextBox").as(LineEdit).text
    onTextEntered(newText)
```

As soon as I use the string interpolation feature  eg the variable story becomes: `var story = &"Once upon a time a {prompts[0]} ate a {prompts[1]} and felt very {prompts[2]}. It was a {prompts[3]} day for {prompts[4]}"` it throws the error `can't find prompts identifier` . 

Now the most quirky thing I don't understand if I change the fields section to:
```nim
  var
    storyText: RichTextLabel
    textBox: LineEdit
    prompts = @["Rishi", "lice", "bad", "great", "ponies"]
    story = &"Once upon a time a {prompts[0]} ate a {prompts[1]} and felt very {prompts[2]}. It was a {prompts[3]} day for {prompts[4]}"
```
I get can't find identifier `story` in the ready method and can't find identifier `textBox` in the `onTextEntered` method.
Am I understanding something wrong here about the gdobj fields?

**Comment by endragor** [comment]
The second thing (multiple declarations in single `var` block) not working was a bug, which I fixed in fd5f77e7c2ce68f17d5b50fc273b54112bf5aa55.

For `&` invocation to work you have to explicitly specify type for `story` variable:
```nim
var story: string = &"Once upon a time a {prompts[0]} ate a {prompts[1]} and felt very {prompts[2]}. It was a {prompts[3]} day for {prompts[4]}"
```

The error occurs because behind the scenes `gdobj` generates type definition that looks like:

```nim
type
  Loony = ref object of Node2D
    storyText: RichTextLabel
    textBox: LineEdit
    prompts: type(@["Rishi", "lice", "bad", "great", "ponies"])
    story: type(&"Once upon a time a {prompts[0]} ate a {prompts[1]} and felt very {prompts[2]}. It was a {prompts[3]} day for {prompts[4]}")
      ## prompts do not exist in this context, so it errors out
```

If you specify type explicitly, it will generate `story: string` there, so no error will occur.

**Comment by zetashift** [comment]
Thanks a lot! Hopefully by the end of the week I get some more small games done:)
